src/julia-syntax.scm

```diff
@@ -1609,6 +1609,16 @@
        (error (string "invalid assignment location \"" (deparse lhs) "\"")))
       (else
        (case (car lhs)
+         ((globalref)
+          ;; M.b =
+          (let* ((a   (cadr lhs))
+                 (b   (caddr lhs))
+                 (rhs (caddr e))
+                 (rr  (if (or (symbol-like? rhs) (atom? rhs)) rhs (make-ssavalue))))
+            `(block
+              ,.(if (eq? rr rhs) '() `((= ,rr ,(expand-forms rhs))))
+              (call (core setfield!) ,a (quote ,b) ,rr)
```

Comment:
it should be valid to lower this line to `(= ,lhs ,rr)`. that should save us from needing to change the meaning of setfield. it looks like it should be handled correctly since https://github.com/JuliaLang/julia/issues/10403